transaction/core/src/encrypted_fog_hint.rs

```diff
@@ -26,9 +26,9 @@ use rand_core::{CryptoRng, RngCore};
 use serde::{Deserialize, Serialize};
 
 // The length of the encrypted fog hint field in the ledger.
-// Must be at least as large as McRistrettoBox::FooterSize, or it can't hold a
-// Ristretto-Box encryption.
-pub type EncryptedFogHintSize = U128;
+// Must be at least as large as mc_crypto_box::VersionedCryptoBox::FooterSize.
+// Footersize = 50, + 32 for one curve point, + 8 bytes of magic / padding space for future needs
```

Comment:
I suggest we drop the padding. I don't see what it is useful for and we can version a new transaction type later if we need more space... so `U82`?

Reply:
The padding would allow us to make a change where, based on the value of the padding, fog treats something differently, in constant time

Maybe just 2 bytes? U84?

Reply:
idk, maybe it's pointless, i'm not sure what it would do differently

Reply:
@rjwalters i think 2 bytes is not a terrible price to pay, and it might give us another option to solve some problem later without creating breaking changes. i guess i have in mind some like compliance thing where we can't store some of the users' rng's in sgx the way want to, or must use different code or something. like basically to accommodate a need that isn't motivated by engineering

Reply:
it seems like this adds a "version" field to fog but I don't think in practice we would need to add a version code to the public records and then ingest wouldn't have any reason to hide it... we should collect other opinions, but I am leaning towards `U82`

Reply:
@rjwalters i guess i'm at a point now where its like, i have a few other things i'm trying to do before code freeze, and deleting these magic bytes completely is more work than just reducing it to two magic bytes, so from that point of view, i like this revision. but i also don't really want to fight over two bytes. but at the same time, no one else was saying they think we should get rid of these two bytes.

so at this point i would like to just merge with U84 and do other things. let me know if that sounds alright to you

Reply:
I know you have a lot on your plate... Maybe leave this open for now and see if you can find time to remove the magic bytes after you finish up the other stuff?

This is the sort of cruft that ends up staying around forever.

Reply:
In general, I'm happy to see future-proofing efforts, especially if they are low-cost. On the other hand, I can see the wariness of prematurely laying plumbing for something without a clear need for it.  

I would vote for keeping the two magic bytes, will seek another opinion from the team as well.